house/house.go

```diff
@@ -0,0 +1,378 @@
+				"ip":   item.Id.Ip,
+			},
+		})
+		metrics[item.Name+"_lastseen"] = promauto.NewGauge(prometheus.GaugeOpts{
+			Name: "home_detector_device_lastseen",
+			Help: "Device in home last seen",
+			ConstLabels: prometheus.Labels{
+				"name": strings.ReplaceAll(item.Name, " ", "_"),
+				"mac":  item.Id.Mac,
+				"ip":   item.Id.Ip,
+			},
+		})
+	}
+}
+
+func (s *Server) recordMetrics() {
+	go func() {
+		for {
+			homeCounter := 0
+			for _, person := range houseDevices {
+				if !person.Away && person.Person {
+					homeCounter++
+				}
+			}
+			peopleHome.Set(float64(homeCounter))
+
+			for _, item := range append(iotDevices, houseDevices...) {
+				state := 1
+				if item.Away {
+					state = 0
+				}
+				if metrics[item.Name] != nil {
+					metrics[item.Name].Set(float64(state))
+				}
+				if metrics[item.Name+"_lastseen"] != nil {
+					metrics[item.Name+"_lastseen"].Set(float64(item.LastSeen))
+				}
+			}
+			time.Sleep(2 * time.Second)
+		}
+	}()
+}
+func (s *Server) adjustLights(lightGroup string, brightness string) error {
+	_, err := s.callAssistant("set " + lightGroup + " lights to " + brightness + "% brightness")
+	if err != nil {
+		return err
+	}
+	return nil
+}
+
+func (s *Server) callAssistant(command string) (*string, error) {
+	if *debug {
+		return &command, nil
+	}
+	return assistant.Call(command)
+}
+
+func (s *Server) deviceDetectState(item device) int64 {
+	lastSeen := int64(time.Now().Unix()) - item.LastSeen
+	return lastSeen
+}
+
+func (s *Server) newDevice(in *pb.AddressRequest) error {
+	name := in.Ip
+	if in.Mac != "" {
+		name = in.Mac
+	}
+	newDevice := device{
+		Name:     strings.ReplaceAll(strings.ReplaceAll(name, ".", "_"), ":", "_"),
+		Id:       networkId{Ip: in.Ip, Mac: in.Mac},
+		Away:     false,
+		LastSeen: int64(time.Now().Unix()),
+		Person:   false,
+		Command:  "",
+	}
+
+	log.Println(fmt.Printf("New Device: %s", name))
+	if !*debug {
+		err := notifications.SendNotification(newDevice.Name, "New Device")
+		if err != nil {
+			return err
+		}
+	}
+
+	houseDevices = append(houseDevices, &newDevice)
+	err := writeConfig("config/devices.yaml")
+	if err != nil {
+		return err
+	}
+	s.registerMetric(newDevice)
+	return nil
+}
+
+func (s *Server) existingDevice(houseDevice *device) error {
+	log.Println(houseDevice.Name)
+	timeAway := s.deviceDetectState(*houseDevice)
+	if timeAway > *timeAwaySeconds && houseDevice.Person {
+		log.Println(fmt.Sprintf("Device: %s has returned after %d seconds", houseDevice.Name, timeAway))
+		if *debug {
+			log.Printf("Notification: %s, %s", houseDevice.Name, "is home")
+		} else {
+			err := notifications.SendNotification(houseDevice.Name, "has returned home.")
+			if err != nil {
+				return err
+			}
+		}
+
+	}
+	houseDevice.Away = false
+	houseDevice.LastSeen = int64(time.Now().Unix())
+	return nil
+}
+
+// Address Handler for receiving IP/MAC requests
+func (s *Server) Address(ctx context.Context, in *pb.AddressRequest) (*pb.Reply, error) {
```

Comment:
warning: cyclomatic complexity 17 of function (*Server).Address() is high (> 15) (gocyclo)